### PR TITLE
Event creation: event viewModel

### PR DESCRIPTION
**Pull request by unglazedstamp**
The event viewModel for task #23.
It's the minimal version needed for the event creation composable to work.

### CODE REVIEW

**Review by alejandrocalles — APPROVED** [review]
Overall an excellent PR. I have made some very small suggestions that you may want to apply before merging, but they are not essential. Thanks for your work!

**Review by srsingh04 — CHANGES_REQUESTED** [review]
The code is well written, error handling is also well done. Just put the text in the strings file as Gabriel said and you are good to go :)

File: app/src/main/java/com/github/swent/echo/viewmodels/event/EventViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+            }
+        }
+    }
+
+    /**
+     * if the tagName is in the global tag list, add it as a tag to the event tag list return a tag
+     * if the string match its name or null otherwise
+     */
+    fun getAndAddTagFromString(tagName: String): Tag? {
+        val tag = allTagsList.filter { (_, name) -> name == tagName }
```
[the remark this answers is not in the document]
Comment (unglazedstamp) [reply app/src/main/java/com/github/swent/echo/viewmodels/event/EventViewModel.kt]: Yes, that's a good idea

File: app/src/test/java/com/github/swent/echo/viewmodels/event/EventViewModelTest.kt [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    fun addTagToEventTest() {
+        val eventViewModel = EventViewModel()
+        val newTag = Tag("2", "tag2")
+        val addedTag = eventViewModel.getAndAddTagFromString(newTag.name)
+        assertEquals(addedTag, newTag)
+        assertEquals(eventViewModel.getEvent().tags, setOf(newTag))
+    }
+
+    @Test
+    fun deleteTagOfEventTest() {
```
[the remark this answers is not in the document]
Comment (unglazedstamp) [reply app/src/test/java/com/github/swent/echo/viewmodels/event/EventViewModelTest.kt]: Thanks, I changed it

**Review by srsingh04 — APPROVED** [review]
All perfect!

**Pull request merged**